dms/models/directory.py

```diff
@@ -643,17 +644,23 @@ def write(self, vals):
             res = super().write(vals)
 
         if self and any(
-            field for field in vals if field in ["root_storage_id", "parent_id"]
+            field for field in vals if field in ["storage_id", "parent_id"]
         ):
             records = self.sudo().search([("id", "child_of", self.ids)]) - self
-            if "root_storage_id" in vals:
-                records.write({"storage_id": vals["root_storage_id"]})
-            elif "parent_id" in vals:
+            if "parent_id" in vals:
```

Comment:
You should prevent the same to change the parent if the target is from another storage.